crates/nostr/src/event/mod.rs

```diff
@@ -35,7 +35,10 @@ pub enum Error {
     Secp256k1(#[from] secp256k1::Error),
     /// Hex decoding error
     #[error(transparent)]
-    Hex(#[from] bitcoin_hashes::hex::Error),
+    Hex(#[from] bitcoin::hashes::hex::Error),
```

Comment:
Change this to `bitcoin_hashes::hex::Error`

Reply:
Changing so that the lib is coherent.

However, I think a following PR is needed:

1) use `bitcoin_hashes` via the re-export `bitcoin::hashes` everywhere
2) remove `bitcoin_hashes` from `Cargo.toml`

This would allow to never have multiple versions of `bitcoin_hashes

Reply:
Yeah, I know. 
In the past versions I used bitcoin as "main" dependency (include both bech32, bitcoin_hashes and secp256k1 deps).
But contains also some code that is not necessary to nostr (except for nip06, but it's a feature so can be disabled).

For this reason, not long ago, I decided to set bitcoin as optional dep.
To avoid compilation of multiple versions of these deps I always check/use the same versions used in bitcoin crate. 

If there is a better way, I'm open to suggestions.

Reply:
Oh sorry, I missed that bitcoin dep is optional, then no I have no suggestion for a better way

Reply:
The compilation it's a little bit slow using the `bitcoin` crate. But I think that this in not a big problem.

Tests done with default features set to `false`:
nostr `v0.18`: 2m and 27s 
new nostr version with `bitcoin` as optional dep: 34s

I decided to use `bitcoin` as optional dep only to optimize the binary size. For example if a dev want to make an app without default features (so only for key generation in an offline environment) the `bitcoin` crate code is not needed. 

But doing some tests just now, the results of the binary size seems to be the same. Seems strange since the size of `bitcoin` dep is 504kb.
I'll do some more tests.

Reply:
Generally the linker is very good at removing unused code path

Reply:
My bad, I should have done these tests before...
So maybe it's better to revert [`ecb96c9`](https://github.com/rust-nostr/nostr/commit/ecb96c9082566a1965d14cf829d70c38e3e4b0b1) commit before release the new version, to avoid to use multiple versions of `secp256k1`, `bech32` or `bitcoin_hashes`.